lib/utils/googleAuth.spec.ts

```diff
@@ -0,0 +1,483 @@
+        };
+
+        const session = getSession(req);
+        const expectedSession: HackneyGoogleUserWithPermissions = {
+          ...tokenData,
+          ...getClaimsByRole(UserRole.Admin),
+        };
+        expect(session).toStrictEqual(expectedSession);
+      });
+
+      it('returns matching authorization groups for manager claims', () => {
+        const { signedToken, tokenData } = generateSignedTokenByRole(
+          UserRole.Manager
+        );
+
+        req.headers = {
+          cookie: `hackneyToken=${signedToken}`,
+        };
+
+        const session = getSession(req);
+        const expectedSession: HackneyGoogleUserWithPermissions = {
+          ...tokenData,
+          ...getClaimsByRole(UserRole.Manager),
+        };
+        expect(session).toEqual(expectedSession);
+      });
+
+      it('returns matching authorization groups for officer claims', () => {
+        const { signedToken, tokenData } = generateSignedTokenByRole(
+          UserRole.Officer
+        );
+
+        req.headers = {
+          cookie: `hackneyToken=${signedToken}`,
+        };
+
+        const session = getSession(req);
+        const expectedSession: HackneyGoogleUserWithPermissions = {
+          ...tokenData,
+          ...getClaimsByRole(UserRole.Officer),
+        };
+        expect(session).toEqual(expectedSession);
+      });
+    });
+  });
+
+  describe('removeHackneyToken', () => {
+    it('sets correct Set-Cookie header', () => {
+      const expectedCookieHeader =
+        'hackneyToken=; Max-Age=-1; Domain=.hackney.gov.uk; Path=/';
+      const res: ApiResponse = createResponse();
+
+      removeHackneyToken(res);
+
+      const newCookieHeader = res.getHeader('Set-Cookie');
+      expect(newCookieHeader).toBe(expectedCookieHeader);
+    });
+  });
+
+  describe('getPermissions', () => {
+    it("returns permissions based on user's group claims", () => {
+      const adminGroupFixture = envVarsFixture('AUTHORISED_ADMIN_GROUP');
+      const managerGroupFixture = envVarsFixture('AUTHORISED_MANAGER_GROUP');
+      const officerGroupFixture = envVarsFixture('AUTHORISED_OFFICER_GROUP');
+      const readOnlyGroupFixture = envVarsFixture('AUTHORISED_READONLY_GROUP');
+      adminGroupFixture.mock(AUTHORISED_ADMIN_GROUP_TEST);
+      managerGroupFixture.mock(AUTHORISED_MANAGER_GROUP_TEST);
+      officerGroupFixture.mock(AUTHORISED_OFFICER_GROUP_TEST);
+      readOnlyGroupFixture.mock(AUTHORISED_READONLY_GROUP_TEST);
+
+      const groupClaims = [
+        AUTHORISED_ADMIN_GROUP_TEST,
+        AUTHORISED_MANAGER_GROUP_TEST,
+        AUTHORISED_OFFICER_GROUP_TEST,
+        AUTHORISED_READONLY_GROUP_TEST,
+      ];
+
+      const user = generateJWTTokenTestData(groupClaims, issuedAt);
+
+      const expectedPermissions: Permissions = {
+        hasAdminPermissions: true,
+        hasManagerPermissions: true,
+        hasOfficerPermissions: true,
+        hasReadOnlyPermissions: true,
+      };
```

Comment:
Can the user have more than 1 set of permissions?

Reply:
Technicaly they could be in multiple groups, but of course that's not how the permissions are setup in practise. However this particular test is just for checking that the claims are set correctly based on the provided groups.